File: solutions/764.n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null) return res;
        Queue<Node> queue = new LinkedList<>();
        List<Integer> curRes = null;
        Node curNode = null;
        queue.offer(root); 
        while(!queue.isEmpty()) {
            curRes = new ArrayList<>();
            for(int size = queue.size(); size > 0; size--) {
                curNode = queue.remove();
                curRes.add(curNode.val);
                for(Node node : curNode.children) queue.offer(node);
            }
            res.add(curRes);
        }
        return res;
    }
}

File: solutions/764.n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        dfs(root, 0, res);
        return res;
    }
private:
    void dfs(Node* root, int depth, vector<vector<int>>& res) {
        if(!root) return;
        if(depth < res.size()) res[depth].push_back(root->val);
        else res.push_back({root->val});
        for(Node* child : root->children) 
            dfs(child, depth + 1, res);
    } 
};